worlds/undertale/__init__.py

```diff
@@ -75,6 +75,12 @@ def _get_undertale_data(self):
             "rando_item_button": bool(self.multiworld.rando_item_button[self.player].value)
         }
 
+    def pre_fill(self):
+        if not self.multiworld.get_location("Starting Key", self.player).item:
+            chosen_key_start = self.random.choice(["Ruins Key", "Snowdin Key", "Waterfall Key", "Hotland Key"])
+            self.multiworld.get_location("Starting Key", self.player).place_locked_item(self.create_item(chosen_key_start))
+            self.multiworld.itempool.remove(self.create_item(chosen_key_start))
+
```

Comment:
This method of solving the key issue has a couple issues.

1. This doesn't allow keys to be item linked.
2. Removing from the `multiworld.itempool` is a bad idea because it's a costly operation.

It would be better to move this logic to `create_items` and not add the item that is going to be `place_locked_item` to the item pool to begin with.

Reply:
But then it wouldn't be plandoable, which is something I want.